pkg/apiserver/apiserver.go

```diff
@@ -241,12 +244,55 @@ func (s *APIServer) Router() (*gin.Engine, error) {
 	return s.router, nil
 }
 
+func (s *APIServer) GetTLSConfig() (*tls.Config, error) {
+	var caCert []byte
+	var err error
+	var caCertPool *x509.CertPool
+	var clientAuthType tls.ClientAuthType
+
+	if s.TLS == nil {
+		return &tls.Config{}, nil
+	}
+
+	if s.TLS.ClientVerification == nil {
+		//sounds like a sane default : verify client cert if given, but don't make it mandatory
+		clientAuthType = tls.VerifyClientCertIfGiven
+	} else {
+		clientAuthType = tls.ClientAuthType(*s.TLS.ClientVerification)
+	}
+	if clientAuthType > tls.RequestClientCert {
+		log.Infof("(tls) Client Auth Type set to %s", clientAuthType.String())
+		caCert, err = ioutil.ReadFile(s.TLS.CACertPath)
```

Comment:
change a bit the logic here: currently crowdsec won't start if no `ca_cert_path` is given, but this is optional (we default to VerifyClientCertIfGiven which does not force client auth)